### ISSUE TITLE
Run tests on Safari 9

### ISSUE THREAD

**Issue by Narretz** [angular/angular.js#13706]
We currently only run Safari 8 on OS X 10.10 and Safari 8.1 on the iPhone.

In Safari 9, $log throws errors:

```
Safari 9.0.0 (Mac OS X 10.11) $log $log.error should pass error if does not have trace FAILED
    TypeError: Attempted to assign to readonly property. in /home/travis/build/angular/angular.js/test/ng/logSpec.js (line 144)
    /home/travis/build/angular/angular.js/test/ng/logSpec.js:144:8
    TypeError: undefined is not an object (evaluating '$log.error') in /home/travis/build/angular/angular.js/test/ng/logSpec.js (line 155)
    /home/travis/build/angular/angular.js/test/ng/logSpec.js:155:11
Safari 9.0.0 (Mac OS X 10.11) $log $log.error should print stack FAILED
    TypeError: Attempted to assign to readonly property. in /home/travis/build/angular/angular.js/test/ng/logSpec.js (line 144)
    /home/travis/build/angular/angular.js/test/ng/logSpec.js:144:8
    TypeError: undefined is not an object (evaluating '$log.error') in /home/travis/build/angular/angular.js/test/ng/logSpec.js (line 162)
    /home/travis/build/angular/angular.js/test/ng/logSpec.js:162:11
Safari 9.0.0 (Mac OS X 10.11) $log $log.error should print line FAILED
    TypeError: Attempted to assign to readonly property. in /home/travis/build/angular/angular.js/test/ng/logSpec.js (line 144)
    /home/travis/build/angular/angular.js/test/ng/logSpec.js:144:8
    TypeError: Attempted to assign to readonly property. in /home/travis/build/angular/angular.js/test/ng/logSpec.js (line 169)
    /home/travis/build/angular/angular.js/test/ng/logSpec.js:169:8
```

**Comment by lgalfaso** [comment]
It looks like Safari handles the properties `sourceURL` and `line` as non-deletable/read-only. Anyhow, we do not really need `e` to be an `Error` as just replacing logSpec.js@141 to
```js
function MockError() {}
MockError.prototype = Error.prototype;
e = new MockError('');
```
makes Safari happy

**Comment by petebacondarwin** [comment]
Fixed the `$log` errors independently by a1ff35850c213d4da5cb11f214c5c917ca0639cf
But we will still have problems with ES6 classes on Safari 9
